src/ripple/basics/impl/make_SSLContext.cpp

```diff
@@ -77,7 +77,21 @@ using rsa_ptr = custom_delete_unique_ptr <RSA>;
 
 static rsa_ptr rsa_generate_key (int n_bits)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x00908000L
+    BIGNUM *bn = BN_new();
+    BN_set_word(bn, RSA_F4);
+
+    RSA* rsa = RSA_new();
+    if (RSA_generate_key_ex(rsa, n_bits, bn, nullptr) != 1)
+    {
+        RSA_free(rsa);
+        rsa = nullptr;
+    }
+
+    BN_free(bn);
+#else
     RSA* rsa = RSA_generate_key (n_bits, RSA_F4, nullptr, nullptr);
+#endif
```

Comment:
Suggested way to hold these resources:  https://github.com/HowardHinnant/rippled/commit/720aa290c221a61797a5a960c3fa83fc539f4b53